Sources/SourceControl/GitRepository.swift

```diff
@@ -63,7 +63,7 @@ public struct GitRepositoryProvider: RepositoryProvider {
     }
 
     @discardableResult
-    private func callGit(_ args: String...,
+    public func callGit(_ args: String...,
```

Comment:
instead of making this public, how about we add a new specific public function? u.e.

```
internal func isGitDirecotry(path: AbsolutePath) throws -> Bool {
        self.queue.sync {
            do {
                let output = try callGit("rev-parse", "--git-dir", "\(path.pathString)"
                                         failureMessage: "Couldn’t test if a git directory")
                return true
            }  catch let error as GitCloneError {
                return false
            }
        }
    }
```

may not compile but you get the idea

Reply:
As per my earlier comment, I think it would make sense to make this even more generic, e.g. `verifyRepository` and use it in a generic way from package loading.

Reply:
great point, we should add it to the abstraction then